### ISSUE TITLE
[Power BI] Join cost/pricesheet to populate ListUnitPrice and ListCost when missing

### ISSUE THREAD

**Issue by flanakin** [microsoft/finops-toolkit#615]
## ⚠️ Problem
Cost Management does not include x_ListUnitPrice and x_ListCost for all records.

## 🛠️ Solution
Join the cost data with the public pricesheet and calculate the cost by multiplying the price by PricingQuantity.

## ℹ️ Additional context
This workaround will slow down report generation. If you are faced with this issue, please file a support request to help raise the priority for the Cost Management team to resolve the underlying data issue.

## 🙋‍♀️ Ask for the community

We could use your help:
1. Please vote this issue up (👍) to prioritize it.
2. Leave comments to help us solidify the vision.
3. File a support request with the Cost Management team to ensure these values are always populated correctly.

**Comment by Kibastola** [comment]
Please could you provide instructions how to **"Join the cost data with the public pricesheet and calculate the cost by multiplying the price by PricingQuantity."**? Also what do you mean by "File a support request with the Cost Management team" ? what is the contact channel for that team?

**Comment by RolandKrummenacher** [comment]
My observation is that ListUnitPrice and ListCost for reservations are always 0 while they seems to be correct for savings plan. Is this related to this issue or is this by design and I'm missing something?

**Comment by RolandKrummenacher** [comment]
What would be the join statement? MeterId and Region?

**Comment by flanakin** [comment]
@Kibastola joining datasets requires pulling in the price data and joining them based on a few columns that depend on the account type. I don't have the exact columns today, but we will get them as part of this effort.

Filing a support request is about raising awareness with the Cost Management team that this is important to you. You can do this in the Azure portal. You can also escalate it with your account team.

**Comment by flanakin** [comment]
> My observation is that ListUnitPrice and ListCost for reservations are always 0 while they seems to be correct for savings plan. Is this related to this issue or is this by design and I'm missing something?

@RolandKrummenacher correct. This is a gap in Cost Management. The team is aware of the issue, but does not have an ETA for when it will be resolved. Doing the join in the toolkit is about implementing a workaround in the meantime.

**Comment by flanakin** [comment]
Note for FTEs: The Cost Management team is tracking this with work item 17121858 internally.
